packages/financial-templates-lib/src/logger/PagerDutyTransportV2.ts

```diff
@@ -0,0 +1,49 @@
+// This transport enables winston logging to send messages to pager duty v2 api.
+import Transport from "winston-transport";
+import { event } from "@pagerduty/pdjs";
+
+type TransportOptions = ConstructorParameters<typeof Transport>[0];
+export type Severity = "critical" | "error" | "warning" | "info";
+
+interface Config {
+  integrationKey: string;
+  customServices?: { [key: string]: string };
+}
+
+export class PagerDutyTransportV2 extends Transport {
+  private readonly integrationKey: string;
+  private readonly customServices: { [key: string]: string };
+  constructor(winstonOpts: TransportOptions, { integrationKey, customServices = {} }: Config) {
+    super(winstonOpts);
+    this.integrationKey = integrationKey;
+    this.customServices = customServices;
+  }
+  // pd v2 severity only supports critical, error, warning or info.
+  public static convertLevelToSeverity(level?: string): Severity {
+    if (!level) return "error";
+    if (level === "warn") return "warning";
+    if (level === "info" || level === "critical") return level;
+    return "error";
+  }
+  // Note: info must be any because that's what the base class uses.
+  async log(info: any, callback: () => void): Promise<void> {
+    try {
+      await event({
+        data: {
+          routing_key: this.integrationKey,
+          event_action: "trigger",
+          payload: {
+            summary: `${info.level}: ${info.at} ⭢ ${info.message}`,
+            severity: PagerDutyTransportV2.convertLevelToSeverity(this.level),
+            source: info["bot-identifier"] ? info["bot-identifier"] : undefined,
+            custom_details: info,
```

Comment:
I think you need to do more post-processing on `info` based on it's type. if it is `mrkdwn ` then you likely want to extract just the `mrkdwn ` part. else, you should likely `mrkdwn ` the object (see [here](https://github.com/UMAprotocol/protocol/blob/9fabd6cd7d2f1d0508199f779f64957f950c5f42/packages/financial-templates-lib/src/logger/PagerDutyTransport.ts#L38).)

I would recommend you set up a sample escalation policy that only pages you and then test out your transport with sample code that looks something like this:
```ts
import { delay, Logger } from "@uma/financial-templates-lib";

export async function run(): Promise<void> {
  const logger = Logger;
  // try messages with mrkdwn, some with json objects, some with nested errors.
  logger.error({
    at: "tester",
    message: "hello world",
    mrkdwn:
      "*headding*\nsome test
  });
logger.info({
    at: "tester",
    message: "hello world",
    someComplexType:{a:b}
  });
}

if (require.main === module) {
  run()
    .then(async () => {
      await delay(5);
      process.exit(0);
    })
    .catch(async (error) => {
      console.error("Process exited with", error);
      await delay(5);
      process.exit(1);
    });
}
```

Reply:
tested using a similar script. further parsing of info isnt really necessary, we can include all the data without extra parsing, heres what the markdown looks like:
![image](https://user-images.githubusercontent.com/4429761/188693099-04add179-f5c0-4d06-8c89-d077c394a01e.png)
![image](https://user-images.githubusercontent.com/4429761/188693749-d8ee0b90-781c-4e8a-a7c1-ed3331ba1239.png)

compare this to our custom details with current alerts where the data comes in unstructured (probably because of the json stringification before sending to pd):
![image](https://user-images.githubusercontent.com/4429761/188694126-0a1338b1-6c64-4e1a-ac0b-1148f9c85518.png)